bin/main

```diff
@@ -1,73 +1,73 @@
-# Variable definition area
 #!/usr/bin/env ruby
 
-$board = %w[1 2 3 4 5 6 7 8 9]
+# Variable definition area
+require_relative '../lib/board.rb'
+require_relative '../lib/players.rb'
 
-i = 0
-name_pone = nil
-name_ptwo = nil
-def display_board(board)
-  puts "#{board[0]} | #{board[1]} | #{board[2]}"
-  puts "---------"
-  puts "#{board[3]} | #{board[4]} | #{board[5]}"
-  puts "---------"
-  puts "#{board[6]} | #{board[7]} | #{board[8]} \n"
-end
+board_array = %w[1 2 3 4 5 6 7 8 9]
+board = Board.new(board_array)
 
-def input_request(num)
-  if num.even?
-    puts "it's #{@name_pone} turn"
-    display_board($board)
-    board_input = gets.chomp
-  else
-    puts "it's #{@name_ptwo} turn"
-    display_board($board)
-    board_input = gets.chomp
-  end
-end
+puts 'Welcome to Tic Tac Toe'
+puts "what's player one's name ?"
 
-def welcome
-  puts "Welcome to Tic Tac Toe"
-  puts "what's player one's name ?"
-  @name_pone = gets.chomp
-  puts "what's player two's name ?"
-  @name_ptwo = gets.chomp
-  system('cls') || system('clear')
-end
+player1_name = gets.chomp
+puts "what's player two's name ?"
+
+player2_name = gets.chomp
+players = Players.new(player1_name, player2_name)
+
+class Interface
+  def display_board(board)
+    puts "#{board[0]} | #{board[1]} | #{board[2]}"
+    puts '---------'
+    puts "#{board[3]} | #{board[4]} | #{board[5]}"
+    puts '---------'
+    puts "#{board[6]} | #{board[7]} | #{board[8]} \n"
+  end
 
-def swap_on_board(i, given_number)
-  while $board.any?(given_number)
-    if i.even?
-      $board[given_number.to_i - 1] = 'X'
-    elsif i.odd?
-      $board[given_number.to_i - 1] = 'O'
+  def input_request(num, player1_name, player2_name, board)
+    if num.even?
+      puts "it's #{player1_name} turn"
+    else
+      puts "it's #{player2_name} turn"
     end
```

Comment:
Let's move the logic into the lib folder. We don't want any logic inside the main.rb.